src/common/args.cpp

```diff
@@ -277,6 +278,18 @@ fs::path ArgsManager::GetPathArg(std::string arg, const fs::path& default_value)
     return result.has_filename() ? result : result.parent_path();
 }
 
+fs::path ArgsManager::GetSignetDataDir() const
+{
+    const std::string base_data_dir = BaseParams().DataDir();
+    const std::string signet_challenge_str = GetArg("-signetchallenge", "");
+    if (!signet_challenge_str.empty()) {
+        std::vector<uint8_t> signet_challenge = ParseHex(signet_challenge_str);
+        const auto hash_160 = Hash160(signet_challenge);
+        return fs::PathFromString(base_data_dir + "-" + HexStr(hash_160));
```

Comment:
The approach to use the 8-character message start (i.e. first 4 bytes of the SHA256 hash of the challenge) seems better to me as well.  More concise directory lengths with a reasonably low chance of collision between signets.

nit:  Recommend avoiding using `-` (dash) and instead using something like `_` (underscore) when building signet paths to prevent unexpected issues when including custom signet directory paths in command line arguments.

Reply:
I have updated the PR to use the `first 8 bytes of the hash160` and replaced the `-` with `_` as suggested